src/connection.rs

```diff
@@ -33,12 +74,16 @@ impl KafkaConnection {
         }
     }
 
-    pub fn new(host: &str, timeout_secs: i32) -> Result<KafkaConnection> {
-        let stream = try!(TcpStream::connect(host));
+    pub fn new(host: &str, timeout_secs: i32, security: Option<SecurityConfig>) -> Result<KafkaConnection> {
+        let plain_stream = try!(TcpStream::connect(host));
+        let stream = match security {
+            Some(config) => KafkaStream::Ssl(try!(SslStream::connect(config.ssl(), plain_stream))),
+            None => KafkaStream::Plain(plain_stream),
+        };
```

Comment:
Ah ... now it's necessary to make another clone of the wrapped `ssl` due to its privacy in `SecurityConfig` :/ This is unfortunate, since the second clone (here in `KafkaConnection::new` through `config.ssl()`) is unnecessary I would say. What about accepting a `Option<openssl::ssl::Ssl>` here instead of a `Option<SecurityConfig>`? (This is fully OK from an API perspective since `KafkaConnection` is a crate private type). This would allow the caller (`ConnectionPool`) to clone the ssl setup/configuration only once:

```
// ... ConnectionPool::get_conn
try!(KafkaConnection::new(host, self.timeout,
                          self.security_config
                              .as_ref()
                              .map(|cfg| cfg.0.clone()))));
```

Further, `SecurityConfig` would then not need be `Clone`able and wouldn't need to expose the public `ssl()` "getter" at all. What do you think?